projects/wave-core-new/src/lib/time/time-slider/time-slider.component.ts

```diff
@@ -0,0 +1,282 @@
+            }
+            if (properties.id === 'end') {
+                this.endTime = this.timeline?.getCustomTime(properties.id) as Date;
+            }
+            if (!this.isRange) {
+                this.endTime = this.startTime;
+            }
+            this.changeTime();
+        });
+
+        this.timeline.on('rangechanged', (properties) => {
+            const startWindow = properties.start;
+            const endWindow = properties.end;
+            const windowSize = endWindow - startWindow;
+            this.changeSelectedScale(windowSize);
+        });
+
+        //Connection to the timestream of the project
+        //changes the position of the Timebars according to the Timestream of the project
+        this.subscriptions.push(
+            this.projectService.getTimeStream().subscribe((t) => {
+                this.startTime = t.start.toDate();
+                this.endTime = t.end.toDate();
+                this.changeDetectorRef.markForCheck();
+                if (this.isRange) {
+                    this.timeline?.removeCustomTime('end');
+                    this.isRange = false;
+                }
+                if (t.end.isAfter(t.start)) {
+                    this.timeline?.addCustomTime(t.end.toDate(), 'end');
+                    this.isRange = true;
+                }
+                this.timeline?.setCustomTime(t.start.toDate(), 'start');
+            }),
+        );
+    }
+
+    changeSelectedScale(windowSize: number): void {
+        this.changeDetectorRef.detectChanges();
+        //these scales have to be adjusted, when the component is moved somewhere else with a higher width
+        const yearScale = 1000 * 60 * 60 * 24 * 31 * 22; //22 months in milliseconds
+        const monthScale = 1000 * 60 * 60 * 24 * 15; //15 days in milliseconds
+        const dayScale = 1000 * 60 * 60 * 31; //31 hours in milliseconds
+        if (windowSize > yearScale) this.selectedScale = 'year';
+        else if (windowSize > monthScale) this.selectedScale = 'month';
+        else if (windowSize > dayScale) this.selectedScale = 'day';
+        else this.selectedScale = 'hour';
+    }
+
+    //changes the Timestream of the project according to the Timebars
+    changeTime(): void {
+        this.projectService.getTimeOnce().subscribe(() => {
+            const updatedTime = new Time(moment(this.startTime), moment(this.endTime));
+            this.projectService.setTime(updatedTime);
+        });
+    }
+
+    ngOnDestroy(): void {
+        this.subscriptions.forEach((s) => s.unsubscribe());
+        this.timeline?.off('timechanged');
+    }
+
+    centerTimeline(): void {
+        this.timeline?.moveTo(this.startTime);
+    }
+
+    changeScale(selectedScale: DurationInputArg2): void {
+        this.projectService.getTimeOnce().subscribe((t) => {
+            const startWindow = t.start.clone().subtract(4, selectedScale);
+            const endWindow = t.start.clone().add(4, selectedScale);
```

Comment:
4 seems to be a magic margin numner. Maybe just extract it to another variable.

Reply:
Did you mean something like that? f85c920b6d4902a399447d9121bbe77120f78398